Java/cp/MaximumOfMinimumSubarrays.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Maximum of Minimum for Every Window Size in an Array
+ * Find the maximum of the minimums of every window size in a given 
+ * positive integer array of size n. The size of windows in an array 
+ * vary from 1 to n. 
+ */
+
+import java.io.*;
+import java.util.*; 
+public class MaximumOfMinimumSubarrays {
+
+    public static int[] findMaximumOfMinimums(int[] numArray, int n) {
+
+        //The next[] and prev[] arrays store the indices of next smaller 
+        //and previous smaller integers for each element.  
+        int[] next = new int[n]; 
+        int[] prev = new int[n];
+
+        //If next smaller element is non-existent, then n is stored, if
+        //previous smaller element in non-existent, then -1 is stored. 
+        for(int i=0; i<n; ++i) {
+            next[i] = n; 
+            prev[i] = -1; 
+        }
+
+        //Using stack to get next smaller element
+        ArrayDeque<Integer> indices = new ArrayDeque<>(); 
```

Comment:
**Stack** is already present in Java, no need to make it complex by using **ArrayDeque** Try to make the Program as simple as it can be.
Implement the same with Stack

Reply:
@siddharth25pandey Yes, but actually Stack class in Java is a subclass of Vector, which gives slower performance... And in official Java documentation of ArrayDeque, I read that "This class is likely to be faster than Stack when used as a stack, and faster than LinkedList when used as a queue." 
Shall I still implement the same with Stack?

Reply:
@neha-b2001, for your ease, let it be the same for this PR, but remember, **our focus should always be on how to make the algorithm easily understandable**, not only reducing the time complexity. Also, in this case, not a very big difference in time complexity will occur if you gonna use stack. Anyways leave it for this, but always remember the sole motto.

Reply:
@siddharth25pandey Okay, understood 👍